docs/configuration.md

```diff
@@ -1,492 +1,442 @@
+comes with sane default values out of the box, you should review it exhaustively
+before moving your systems to production.
 
 ## Override specific configuration options
 
-In a typical setup where you run your Registry from the official image, you can specify a configuration variable from the environment by passing `-e` arguments to your `docker run` stanza, or from within a Dockerfile using the `ENV` instruction.
+In a typical setup where you run your Registry from the official image, you can
+specify a configuration variable from the environment by passing `-e` arguments
+to your `docker run` stanza or from within a Dockerfile using the `ENV`
+instruction.
 
 To override a configuration option, create an environment variable named
-`REGISTRY_variable` where *`variable`* is the name of the configuration option
+`REGISTRY_variable` where `variable` is the name of the configuration option
 and the `_` (underscore) represents indention levels. For example, you can
 configure the `rootdirectory` of the `filesystem` storage backend:
 
-    storage:
-      filesystem:
-        rootdirectory: /var/lib/registry
+```none
+storage:
+  filesystem:
+    rootdirectory: /var/lib/registry
+```
 
 To override this value, set an environment variable like this:
 
-    REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY=/somewhere
+```none
+REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY=/somewhere
+```
 
 This variable overrides the `/var/lib/registry` value to the `/somewhere`
 directory.
 
->**NOTE**: It is highly recommended to create a base configuration file with which environment variables can be used to tweak individual values.  Overriding configuration sections with environment variables is not recommended.
+> **Note**: Create a base configuration file with environment variables that can
+> be configured to tweak individual values. Overriding configuration sections
+> with environment variables is not recommended.
 
 ## Overriding the entire configuration file
 
-If the default configuration is not a sound basis for your usage, or if you are having issues overriding keys from the environment, you can specify an alternate YAML configuration file by mounting it as a volume in the container.
+If the default configuration is not a sound basis for your usage, or if you are
+having issues overriding keys from the environment, you can specify an alternate
+YAML configuration file by mounting it as a volume in the container.
 
-Typically, create a new configuration file from scratch, and call it `config.yml`, then:
+Typically, create a new configuration file from scratch,named `config.yml`, then
+specify it in the `docker run` command:
 
-    docker run -d -p 5000:5000 --restart=always --name registry \
-      -v `pwd`/config.yml:/etc/docker/registry/config.yml \
-      registry:2
+```bash
+$ docker run -d -p 5000:5000 --restart=always --name registry \
+             -v `pwd`/config.yml:/etc/docker/registry/config.yml \
```

Comment:
Maybe out of scope for this PR, but we should avoid bind mounting single files into the container, as it can be cause of many issues. It's better to bind-mount a directory _containing_ the configuration file, i.e.;

    mkdir registry-config
    create config.yml inside registry-config
    bind-mount with `-v pwd/registry-config/:/etc/docker/registry/`